sql/core/src/test/resources/test-data/mixed-types1.csv

```diff
@@ -0,0 +1,4 @@
+col_mixed_types
+2012
+1997
+True
```

Comment:
Let's don't bother to create new data file here. You write the test codes as below instead.

```scala
withTempPath { path =>
  Seq("col_mixed_types", "2012", "1997", "True").toDS.write.text(path.getCanonicalPath)
  val df = spark.read.format("csv")
    .option("header", "true")
    .option("inferSchema", "true")
    .load(path.getCanonicalPath)

  assert(data.schema.last == StructField("col_mixed_types", StringType, true))
}
```

Reply:
Ok, It's more clear to see the data in the test.